### PR TITLE
Add ngrx and hadle state

### PR DESCRIPTION
**Pull request by Nuwa-Hub**
[ITP-1405]
[ITP-1401]

[ITP-1405]: https://fcodelabs.atlassian.net/browse/ITP-1405?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-1401]: https://fcodelabs.atlassian.net/browse/ITP-1401?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/app/post-module/components/card-list/card-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+export class CardListComponent {
+  data = [1, 2, 3, 4, 5, 6, 7, 8];
+  isLoading$: Observable<boolean>;
+  error$: Observable<string | null>;
+  posts$: Observable<PostInterface[]>;
+  constructor(private store: Store<AppStateInterface>) {
+    this.isLoading$ = this.store.pipe(select(isLoadingSelector));
+    this.error$ = this.store.pipe(select(errSelector));
+    this.posts$ = this.store.pipe(select(cardsSelector));
+    console.log(this.posts$);
```
Comment (ChamaraWee) [inline app/app/post-module/components/card-list/card-list.component.ts]: remove logs

File: app/app/post-module/components/card-list/card-list.component.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
+.cardscontainer{
+ 
+width: 100%;
+padding: 0%;
+margin: 0%;
+}
+
```
Comment (ChamaraWee) [inline app/app/post-module/components/card-list/card-list.component.scss]: wrong indentation

**Review by ChamaraWee — CHANGES_REQUESTED** [review]

File: src/app/components/card-form/card-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-card-form',
+  templateUrl: './card-form.component.html',
+  styleUrls: ['./card-form.component.scss']
+})
+export class CardFormComponent implements OnInit {
```
Comment (ChamaraWee) [inline src/app/components/card-form/card-form.component.ts:8]: run prettier

File: src/app/containers/sign-in/sign-in.component.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      <div class="form">
+        <label for="user-name" class="usernamelable"> &nbsp;Username </label>
+        <div class="grid-container">
+          <div class="grid-item">
+            <input
+              id="user-name"
+              class="form-content"
+              type="text"
+              name="username"
+
```
Comment (ChamaraWee) [inline src/app/containers/sign-in/sign-in.component.html:19]: spaces ?

File: src/app/components/card-form/card-form.component.ts (hunk 3 above)
Comment (Nuwa-Hub) [reply src/app/components/card-form/card-form.component.ts:8]: done

File: app/app/post-module/components/card-list/card-list.component.scss (hunk 2 above)
Comment (Nuwa-Hub) [reply app/app/post-module/components/card-list/card-list.component.scss]: done

File: app/app/post-module/components/card-list/card-list.component.ts (hunk 1 above)
Comment (Nuwa-Hub) [reply app/app/post-module/components/card-list/card-list.component.ts]: done

File: src/app/containers/sign-in/sign-in.component.html (hunk 4 above)
Comment (Nuwa-Hub) [reply src/app/containers/sign-in/sign-in.component.html:19]: done

**Comment by Nuwa-Hub** [comment]
done

**Pull request closed without merging**